src/main/scala/edu/utah/cs/simba/DataFrame.scala

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2016 by Simba Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package edu.utah.cs.simba
+
+import edu.utah.cs.simba.execution.QueryExecution
+import edu.utah.cs.simba.expression.{InCircleRange, InKNN, InRange, PointWrapper}
+import edu.utah.cs.simba.plans.{DistanceJoin, KNNJoin}
+import edu.utah.cs.simba.spatial.Point
+import org.apache.spark.sql
+import org.apache.spark.sql.catalyst.expressions.{Attribute, Literal}
+import org.apache.spark.sql.catalyst.plans.logical.{Filter, Join, LogicalPlan}
+
+/**
+  * Created by gefei on 2016/11/14.
+  */
+private[simba] class DataFrame (@transient val simbaContext: SimbaContext,
+                              @transient override val queryExecution: QueryExecution
+                              ) extends sql.DataFrame(simbaContext, queryExecution.withIndexedData) {
```

Comment:
BTW, @Skyprophet do you have some idea to implement the `DataFrame` class? Do we have to implement our own `DataFrame`?

I think if we implement our own `DataFrame` (namely `SimbaDataFrame`), we need to deal with the transformation from `RDD[Row]` to `SimbaDataFrame`, and there is a lot of work to do, such as Type Inferring, all these are `private[sql]` in original package, we cannot reuse them directly.